packages/web/components/Landing/OnboardingGuidance/index.tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  Box,
+  Flex,
+  Heading,
+  MetaButton,
+  MetaTile,
+  MetaTileBody,
+  MetaTileHeader,
+  SimpleGrid,
+  Text,
+  VStack,
+} from '@metafam/ds';
+import BabyOctopus from 'assets/quests/baby_octo.png';
+import Octopus from 'assets/quests/octopus.png';
+import SpaceOctopi from 'assets/quests/space_octo.png';
+import { PageContainer } from 'components/Container';
+import { MarkdownViewer } from 'components/MarkdownViewer';
+import { useRouter } from 'next/router';
+import { useEffect, useRef } from 'react';
+
+import roleJson from './guidance.json';
+import { RolePicture } from './RolePicture';
+
+type GuidanceRole = {
+  role: string;
+  details: string[];
+  image: string;
+};
+
+enum Role {
+  CuriousOcto = 'Curious Octo',
+  EngagedOcto = 'Engaged Octo',
+  SpaceOcto = 'Space Octo',
+}
+
+const OnboardingGuidance: React.FC = () => {
+  const router = useRouter();
+  const guidanceRoles = useRef<GuidanceRole[]>([]);
+  useEffect(() => {
+    guidanceRoles.current = roleJson.roles.map(({ role, list }) => {
+      const guidance: GuidanceRole = { role, details: list, image: '' };
+      if (role === Role.CuriousOcto) guidance.image = BabyOctopus.src;
+      if (role === Role.EngagedOcto) guidance.image = Octopus.src;
+      if (role === Role.SpaceOcto) guidance.image = SpaceOctopi.src;
+      return guidance;
+    });
+  }, []);
```

Comment:
is there a particular need to put the data of roles in `json` file? could you just attach the `image` value to each of the role objects like it's done for example in `QuestChainDetails` which you can find here: `packages/web/utils/questChains.ts`?

then you don't need to use the useEffect and there will also be no need to check for the existence of `!!guidanceRoles.current?.length` and `guidanceRoles.current?.map` below. :)